workbench_access.post_update.php

```diff
@@ -96,3 +97,61 @@ function workbench_access_post_update_convert_user_storage_keys(array &$sandbox)
   $sandbox['#finished'] = empty($sandbox['ids']) ? 1 : ($sandbox['count'] - count($sandbox['ids'])) / $sandbox['count'];
   return t('Updated user assigments');
 }
+
+/**
+ * Transform existing role data to new storage.
+ */
+function workbench_access_post_update_section_role_association(&$sandbox) {
+  $schemes = \Drupal::entityTypeManager()->getStorage('access_scheme')->loadMultiple();
+  $storage = \Drupal::service('workbench_access.role_section_storage');
+  $state = \Drupal::state();
+  foreach ($schemes as $scheme) {
+    foreach (\Drupal::entityTypeManager()->getStorage('user_role')->loadMultiple() as $rid => $role) {
+      $potential_ids = [
+        RoleSectionStorageInterface::WORKBENCH_ACCESS_ROLES_STATE_PREFIX . $rid,
+        RoleSectionStorageInterface::WORKBENCH_ACCESS_ROLES_STATE_PREFIX . 'default__' . $rid,
+      ];
+      foreach ($potential_ids as $key) {
+        if ($existing = $state->get($key, FALSE)) {
+          // Save the new roles.
+          $storage->addRole($scheme, $rid, array_values($existing));
+          // Delete the old storage.
+          $state->delete($key);
+        }
+      }
+    }
+  }
+}
+
+/**
+ * Transform existing user data to new storage.
+ */
+function workbench_access_post_update_section_user_association(&$sandbox) {
+  $schemes = \Drupal::entityTypeManager()->getStorage('access_scheme')->loadMultiple();
+  $storage = \Drupal::service('workbench_access.user_section_storage');
+  $user_storage = \Drupal::entityTypeManager()->getStorage('user');
+  if (!isset($sandbox['ids'])) {
+    $sandbox['ids'] = $user_storage
+      ->getQuery()
+      ->exists(WorkbenchAccessManagerInterface::FIELD_NAME)
+      ->execute();
+    $sandbox['count'] = count($sandbox['ids']);
+  }
+  foreach (array_splice($sandbox['ids'], 0, 50) as $id) {
+    $user = $user_storage->load($id);
+    $existing = array_column($user->get(WorkbenchAccessManagerInterface::FIELD_NAME)->getValue(), 'value');
```

Comment:
We need a post update hook to delete `WorkbenchAccessManagerInterface::FIELD_NAME`.

Reply:
@jibran That makes sense. Does this update work against your test data?